### PR TITLE
[project-base] category query now can return ready category seo mix

### PR DESCRIPTION
**Pull request by grossmannmartin**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Category able to return ReadyCategorySeoMix allows more streamlined storefront implementation
|New feature| Yes 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: project-base/app/src/FrontendApi/Resolver/Category/CategoryQuery.php [hunk 1]
```diff
@@ -25,37 +38,94 @@ class CategoryQuery extends BaseCategoryQuery
...
 
-            return $category;
+            if ($entityClass === Category::class) {
+                try {
+                    $category = $this->categoryFacade->getVisibleOnDomainById($this->domain->getId(), $friendlyUrl->getEntityId());
+                } catch (CategoryNotFoundException) {
+                    throw new CategoryNotFoundUserError('Category with URL slug `' . $urlSlug . '` does not exist.');
+                }
+
+                $matchingReadyCategorySeoMix = $this->findMatchingReadyCategorySeoMix($info, $category);
```
Comment (TomasLudvik) [inline project-base/app/src/FrontendApi/Resolver/Category/CategoryQuery.php:109]: This cannot happen, right? Slugs are unique so if it belongs to the category, we should display category.

**Review by TomasLudvik — CHANGES_REQUESTED** [review]

File: project-base/app/src/FrontendApi/Resolver/Category/CategoryQuery.php (hunk 1 above)
Comment (grossmannmartin) [reply project-base/app/src/FrontendApi/Resolver/Category/CategoryQuery.php:109]: The requested behavior is – when you match the filter criteria with the existing SEO mix, you should be redirected to the appropriate SEO mix.

File: project-base/app/src/FrontendApi/Resolver/Category/CategoryQuery.php [hunk 2]
```diff
@@ -25,37 +38,94 @@ class CategoryQuery extends BaseCategoryQuery
...
+                $this->domain->getId(),
+                $urlSlug,
+            );
+
+            if ($friendlyUrl === null) {
+                $modifiedSlug = TransformString::addOrRemoveTrailingSlashFromString($urlSlug);
+                $friendlyUrl = $this->appFriendlyUrlFacade->findByDomainIdAndSlug(
+                    $this->domain->getId(),
+                    $modifiedSlug,
+                );
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply project-base/app/src/FrontendApi/Resolver/Category/CategoryQuery.php:93]: it's the same behavior as it is in `\Shopsys\FrontendApiBundle\Model\FriendlyUrl\FriendlyUrlFacade::getFriendlyUrlByRouteNameAndSlug` – this method is used for resolving the entity for every other *BySlug query. 
Unfortunately, as it needs a route name, it cannot be used here as I accept Category or Seo mix slugs.

**Review by TomasLudvik — APPROVED** [review]

**Comment by grossmannmartin** [comment]
resolved in #2654

**Pull request closed without merging**